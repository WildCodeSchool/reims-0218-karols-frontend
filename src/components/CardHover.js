import React, { Component } from "react"
import {
  Card,
  CardImg,
  CardFooter,
  CardText,
  CardTitle,
  CardImgOverlay,
  Button
} from "reactstrap"

class CardHover extends Component {
  constructor(props) {
    super(props)
    this.handleMouseHover = this.handleMouseHover.bind(this)
    this.state = {
      isHovering: false
    }
  }
  handleMouseHover() {
    this.setState(this.toggleHoverState)
  }
  toggleHoverState(state) {
    return {
      isHovering: !state.isHovering
    }
  }
  render() {
    return (
      <div>
        <div
          onMouseEnter={this.handleMouseHover}
          onMouseLeave={this.handleMouseHover}
        >
          <div>
            <Card inverse>
              <div>
                <CardImg
                  width="100%"
                  src="http://www.lespiedssurterre.fr/wp-content/uploads/2017/09/MissCoiffure-choix.jpg"
                  alt="Card image cap"
                />
                <CardImgOverlay className="text-center">
                  <CardTitle
                    style={{
                      fontSize: "2em"
                    }}
                  >
                    Titre !
                  </CardTitle>
                  {this.state.isHovering && (
                    <CardText
                      style={{
                        paddingTop: "10px",
                        fontSize: "1.5em"
                      }}
                    >
                      Ca fonctionne plut√¥t bien
                    </CardText>
                  )}
                </CardImgOverlay>
              </div>
              <CardFooter>
                <Button color="default" size="lg" block>
                  Choisir ce service
                </Button>
              </CardFooter>
            </Card>
          </div>
        </div>
      </div>
    )
  }
}

export default CardHover
//{this.state.isHovering && (
