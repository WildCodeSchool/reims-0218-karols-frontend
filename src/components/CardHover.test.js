import React from "react"
import ReactDOM from "react-dom"
import CardHover from "./CardHover"
import renderer from "react-test-renderer"

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<CardHover />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it("renders correctly", () => {
  const tree = renderer.create(<CardHover />).toJSON()
  expect(tree).toMatchSnapshot()
})
