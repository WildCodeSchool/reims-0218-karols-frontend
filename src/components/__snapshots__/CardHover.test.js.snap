// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div>
      <div
        className="card text-white"
      >
        <div>
          <img
            alt="Card image cap"
            className="card-img"
            src="http://www.lespiedssurterre.fr/wp-content/uploads/2017/09/MissCoiffure-choix.jpg"
            width="100%"
          />
          <div
            className="text-center card-img-overlay"
          >
            <h5
              className="card-title"
              style={
                Object {
                  "fontSize": "2em",
                }
              }
            >
              Titre !
            </h5>
          </div>
        </div>
        <div
          className="card-footer"
        >
          <button
            className="btn btn-default btn-lg btn-block"
            onClick={[Function]}
            type={undefined}
          >
            Choisir ce service
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
